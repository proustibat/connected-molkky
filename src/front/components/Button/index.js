import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, children, disabled = false }) => (
  <button
    type="button"
    className={`waves-effect waves-light btn btn-large ${disabled ? 'disabled' : ''}`}
    onClick={onClick}
  >
    {children}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default Button;
