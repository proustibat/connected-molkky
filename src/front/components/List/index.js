import React from 'react';
import PropTypes from 'prop-types';
import kebabCase from 'lodash/kebabCase';
import defaultStyle from './style';

const List = ({ title, elements }) => (
  <ul className="collection">
    {title && <li className="collection-header" style={defaultStyle.collectionHeader}><p style={defaultStyle.collectionHeader__content}>{title}</p></li>}
    {elements.map((el) => <li key={kebabCase(el)} className="collection-item">{el}</li>)}
  </ul>
);

List.propTypes = {
  title: PropTypes.string,
  elements: PropTypes.arrayOf(PropTypes.string).isRequired,
};

List.defaultProps = {
  title: null,
};

export default List;
