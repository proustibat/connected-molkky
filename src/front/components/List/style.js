export default {
    collectionHeader: {
        background: 'rgba(0, 0, 0, 0.05)',
        padding: '1rem 1rem 1rem 0'
    },
    collectionHeader__content: {
        borderLeft: '0.5rem solid',
        padding: '0 1rem',
        margin: 0
    }
}