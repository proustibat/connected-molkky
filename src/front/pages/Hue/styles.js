export default {
  containerTextWithButton: {
    display: 'flex',
    alignItems: 'center',
    flexWrap: 'wrap',
  },
  textBeforeButton: {
    marginRight: '1rem',
    textAlign: 'justify',
  },
  smallButtonInText: {
    display: 'inline-block',
    width: 'auto',
    margin: 0,
  },
};
