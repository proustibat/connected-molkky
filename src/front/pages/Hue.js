import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import Button from '../components/Button';
import LoadingBar from '../components/LoadingBar';

export default class Hue extends React.Component {
    static propTypes = {
        title: PropTypes.string.isRequired
    };

    state = {
        isLoading: false,
        ipaddress: null,
        connectData: null,
        rooms: [],
        lights: []
    };

    componentDidMount() {
        if (typeof window !== 'undefined') {
            require('materialize-css');
        }
    }

    requestDiscover = async () => {
        this.setState(() => ({isLoading: true}));
        const result = await fetch('/api/hue/discover', {method: 'GET'})
            .then(response => response.ok ? response.json() : Promise.reject(response.statusText))
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            ipaddress: get(result, 'ipaddress')
        }));
    };

    requestConnect = async () => {
        this.setState(() => ({isLoading: true}));
        const connectData = await fetch('/api/hue/connect', {
            method: 'post',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ipaddress:  this.state.ipaddress})
        })
            .then(response => response.ok ? response.json() : Promise.reject(response.statusText))
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            connectData
        }));
    };

    requestInfo = async () => {
        this.setState(() => ({isLoading: true}));

        const result = await Promise
            .all([
                fetch(`/api/hue/rooms?ipaddress=${this.state.ipaddress}&username=${this.state.connectData.user.username}`, {method: 'get'})
                    .then(response => response.ok ? response.json() : Promise.reject(response.statusText)),
                fetch(`/api/hue/lights?ipaddress=${this.state.ipaddress}&username=${this.state.connectData.user.username}`, {method: 'get'})
                    .then(response => response.ok ? response.json() : Promise.reject(response.statusText))

            ])
            .then(response => {
                return {
                    rooms: response[0] || [],
                    lights: response[1] || []
                };
            })
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            ...result
        }));
    };

    renderConnectData = () => {
        const {connectData: {message, user: {username}}} = this.state;
        return (
            <>
                <p>{message}</p>
                <p>You're connected as {username}</p>
            </>
        );
    };

    renderRooms = () => {
        const {rooms} = this.state;
        return (
            <>
                <p>You have {rooms.length} room{`${rooms.length > 0 ? 's' : ''}`}:</p>
                <ul className="collection">
                    {rooms.map(room => <li className="collection-item">{room.name}</li>)}
                </ul>
            </>
        );
    };

    renderLights = () => {
        const {lights} = this.state;
        return (
            <>
                <p>You have {lights.length} light{`${lights.length > 0 ? 's' : ''}`}:</p>
                <ul className="collection">
                    {lights.map(light => <li className="collection-item">{light.name}</li>)}
                </ul>
            </>
        );
    };

    toastError = error => {
        M && M.toast({html: error, classes: 'red darken-4'});
    };

    render() {
        return (
            <div className="section no-pad-bot">
                <div className="container">
                    {this.state.isLoading && <LoadingBar/>}
                    <h1>{ this.props.title }</h1>

                    <Button onClick={this.requestDiscover} disabled={this.state.isLoading}>Scan Philips Hue System</Button>

                    {this.state.ipaddress && (
                        <>
                            <p>Bridge IP detected: {this.state.ipaddress}</p>

                            <Button onClick={this.requestConnect} disabled={this.state.isLoading}>Connect</Button>
                            {this.state.connectData && this.renderConnectData()}
                        </>
                    )}

                    {this.state.connectData && (
                        <>
                            <Button onClick={this.requestInfo} disabled={this.state.isLoading}>Get information about your home</Button>
                            {this.state.rooms.length > 0 && this.renderRooms()}
                            {this.state.lights.length > 0 && this.renderLights()}
                        </>
                    )}
                </div>
            </div>
        );
    }
}