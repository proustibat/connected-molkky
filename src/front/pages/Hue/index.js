import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import Button from '../../components/Button';
import List from '../../components/List';
import LoadingBar from '../../components/LoadingBar';

export default class Hue extends React.Component {
    static propTypes = {
        title: PropTypes.string
    };

    state = {
        isLoading: false,
        ipaddress: null,
        connectData: null,
        // connectData: {
        //     user: {
        //         username: 'GUn1rnTiYkifD9cAy2d23yFTFZo36ePWzDA4Hgfs'
        //     }
        // },
        rooms: [],
        lights: []
    };

    requestDiscover = async () => {
        this.setState(() => ({isLoading: true}));
        const result = await fetch('/api/hue/discover', {method: 'GET'})
            .then(response => response.ok ? response.json() : Promise.reject(response.statusText))
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            ipaddress: get(result, 'ipaddress')
        }));
    };

    requestConnect = async () => {
        this.setState(() => ({isLoading: true}));
        const connectData = await fetch('/api/hue/connect', {
            method: 'post',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ipaddress:  this.state.ipaddress})
        })
            .then(response => response.ok ? response.json() : Promise.reject(response.statusText))
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            connectData
        }));
    };

    requestInfo = async () => {
        this.setState(() => ({isLoading: true}));

        const result = await Promise
            .all([
                fetch(`/api/hue/rooms?ipaddress=${this.state.ipaddress}&username=${this.state.connectData.user.username}`, {method: 'get'})
                    .then(response => response.ok ? response.json() : Promise.reject(response.statusText)),
                fetch(`/api/hue/lights?ipaddress=${this.state.ipaddress}&username=${this.state.connectData.user.username}`, {method: 'get'})
                    .then(response => response.ok ? response.json() : Promise.reject(response.statusText))

            ])
            .then(response => {
                return {
                    rooms: response[0] || [],
                    lights: response[1] || []
                };
            })
            .catch(this.toastError);

        this.setState(() => ({
            isLoading: false,
            ...result
        }));
    };

    renderConnectData = () => {
        const {connectData: {message, user: {username}}} = this.state;
        return (
            <>
                <p>{message}</p>
                <p>You're connected as {username}</p>
            </>
        );
    };

    toastError = error => {
        if (typeof window !== 'undefined') {
            const {M: MaterializeCSS} = window;
            MaterializeCSS && MaterializeCSS.toast({html: error, classes: 'red darken-4'});
        }
    };

    render() {
        const {rooms, lights} = this.state;
        return (
            <div className="section no-pad-bot">
                <div className="container">
                    {this.state.isLoading && <LoadingBar/>}
                    {this.props.title && <h1>{ this.props.title }</h1>}

                    <Button onClick={this.requestDiscover} disabled={this.state.isLoading}>Scan Philips Hue System</Button>

                    {this.state.ipaddress && (
                        <>
                            <p>Bridge IP detected: {this.state.ipaddress}</p>

                            <Button onClick={this.requestConnect} disabled={this.state.isLoading}>Connect</Button>
                            {this.state.connectData && this.renderConnectData()}
                        </>
                    )}

                    {this.state.connectData && (
                        <>
                            <Button onClick={this.requestInfo} disabled={this.state.isLoading}>Get information about your home</Button>
                            {this.state.rooms.length > 0 && <List title={`You have ${rooms.length} room${rooms.length > 0 ? 's' : ''}:`} elements={rooms.map(room => room.name)} />}
                            {this.state.lights.length > 0 && <List title={`You have ${lights.length} light${rooms.length > 0 ? 's' : ''}:`} elements={lights.map(light => light.name)} />}
                        </>
                    )}
                </div>
            </div>
        );
    }
}