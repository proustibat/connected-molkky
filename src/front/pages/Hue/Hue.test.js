import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Button';
import {
  ipaddress, connectData, lights, rooms,
} from '../../../fixtures/hue';
import List from '../../components/List';
import Hue from './index';

const roomsEndpoint = '/api/hue/rooms';
const lightsEndpoint = '/api/hue/lights';
const expectedHeaders = { Accept: 'application/json', Authorization: `Bearer ${connectData.token}` };

describe('Page Hue', () => {
  it('should render the page correctly on init', () => {
    // Given / When
    const component = shallow(<Hue />);

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should render the page correctly with all props and state data', () => {
    // Given / When
    const component = shallow(<Hue title="Hello World!" />);
    component.setState({
      isLoading: false,
      ipaddress,
      token: connectData.token,
      lights,
      rooms,
      ipaddressFromStorage: false,
    });

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should get data from local storage on mount and update the state', async () => {
    // Given
    jest.spyOn(Storage.prototype, 'getItem').mockImplementation((key) => ({
      token: connectData.token,
      ipaddress,
    }[key]));

    // When
    const component = await shallow(<Hue />);

    // Then
    expect(localStorage.getItem).toHaveBeenCalledTimes(2);
    expect(localStorage.getItem.mock.calls[0]).toEqual(['token']);
    expect(localStorage.getItem.mock.calls[1]).toEqual(['ipaddress']);
    expect(component.state().token).toBe(connectData.token);
    expect(component.state().ipaddress).toBe(ipaddress);

    localStorage.getItem.mockRestore();
  });

  it('should clear the local storage if there is a token in it but no ipaddress', async () => {
    // Given
    jest.spyOn(Storage.prototype, 'getItem').mockImplementation((key) => ({
      token: connectData.token,
    }[key]));
    jest.spyOn(Storage.prototype, 'clear');

    // When
    const component = await shallow(<Hue />);

    // Then
    expect(localStorage.getItem).toHaveBeenCalledTimes(2);
    expect(localStorage.getItem.mock.calls[0]).toEqual(['token']);
    expect(localStorage.getItem.mock.calls[1]).toEqual(['ipaddress']);
    expect(component.state().token).toBeNull();
    expect(component.state().ipaddress).toBeNull();
    expect(localStorage.clear).toHaveBeenCalledTimes(1);

    localStorage.getItem.mockRestore();
  });

  describe('Given the api requests succeed', () => {
    let component;
    let componentInstance;
    let setStateSpy;

    beforeEach(() => {
      component = shallow(<Hue title="Hello World!" />);
      componentInstance = component.instance();
      component.setState({ ipaddressFromStorage: false });
      setStateSpy = jest.spyOn(componentInstance, 'setState');
    });

    afterEach(() => {
      setStateSpy.mockClear();
    });

    afterAll(() => {
      global.fetch.mockRestore();
      delete global.fetch;
      setStateSpy.mockRestore();
    });

    it('should handle discover request', (done) => {
      // Given
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve({ ipaddress }),
      }));

      // When
      component.find(Button).at(0).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenLastCalledWith('/api/hue/discover', { method: 'GET' });
        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();
        expect(component.state().ipaddress).toBe(ipaddress);

        expect(component).toMatchSnapshot('Render After Discover');

        done();
      });
    });

    it('should handle connect request', (done) => {
      // Given
      component.setState({ ipaddress, token: null });
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(connectData),
      }));

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenLastCalledWith(
          '/api/hue/connect',
          {
            method: 'post',
            headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
            body: '{"ipaddress":"192.168.1.10"}',
          },
        );
        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();
        expect(component.state().token).toBe(connectData.token);
        expect(component.state().ipaddressFromStorage).toBeFalsy();

        expect(component).toMatchSnapshot('Render After Connect');

        done();
      });
    });

    it('should save data after connect request', (done) => {
      // Given
      component.setState({ ipaddress, token: null });
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(connectData),
      }));
      jest.spyOn(Storage.prototype, 'setItem');

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      setImmediate(() => {
        expect(localStorage.setItem).toHaveBeenCalledTimes(2);
        expect(localStorage.setItem).toHaveBeenNthCalledWith(1, 'token', 'u-n-i-t_t-e-s-t_t-o-k-e-n');
        expect(localStorage.setItem).toHaveBeenNthCalledWith(2, 'ipaddress', '192.168.1.10');
        localStorage.setItem.mockRestore();
        done();
      });
    });

    it('should handle info request', (done) => {
      // Given
      component.setState({ ipaddress, token: connectData.token });
      global.fetch = jest.fn().mockImplementation((endpoint) => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(endpoint === roomsEndpoint ? rooms : lights),
      }));

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        expect(global.fetch).toHaveBeenCalledTimes(2);
        expect(global.fetch).toHaveBeenNthCalledWith(1, roomsEndpoint, { method: 'get', headers: expectedHeaders });
        expect(global.fetch).toHaveBeenNthCalledWith(2, lightsEndpoint, { method: 'get', headers: expectedHeaders });
        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();
        expect(component.state().rooms).toBe(rooms);
        expect(component.state().lights).toBe(lights);

        expect([
          component.find(List).at(0).props().title,
          component.find(List).at(1).props().title,
        ]).toMatchSnapshot();

        expect(component).toMatchSnapshot('Render After Info Request');

        done();
      });
    });

    it('should render the information lists correctly when there is no element', (done) => {
      // Given
      component.setState({ ipaddress, token: connectData.token });
      const mockedResponse = { ok: true, json: () => Promise.resolve(null) };
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve(mockedResponse));

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      setImmediate(() => {
        expect([
          component.find(List).at(0).props().title,
          component.find(List).at(1).props().title,
        ]).toMatchSnapshot();
        done();
      });
    });

    it('should render the information lists correctly when there is only one element', (done) => {
      // Given
      component.setState({ ipaddress, token: connectData.token });
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve({ ok: true, json: () => Promise.resolve(['single element']) }));

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      setImmediate(() => {
        expect([
          component.find(List).at(0).props().title,
          component.find(List).at(1).props().title,
        ]).toMatchSnapshot();
        done();
      });
    });
  });

  describe('Given the api requests fail', () => {
    const errorMessage = 'unit test error';
    let toastSpy; let component; let componentInstance; let setStateSpy;

    beforeAll(() => {
      const mockedResponse = { ok: false, statusText: errorMessage };
      global.fetch = jest.fn().mockImplementation(() => Promise.resolve(mockedResponse));
      toastSpy = jest.fn();
      global.M = { toast: toastSpy };
    });

    beforeEach(() => {
      // Given
      component = shallow(<Hue title="Hello World!" />);
      component.setState({ ipaddressFromStorage: false });
      componentInstance = component.instance();
      setStateSpy = jest.spyOn(componentInstance, 'setState');
    });

    afterEach(() => {
      setStateSpy.mockRestore();
      global.fetch.mockClear();
      global.M.toast.mockClear();
    });

    afterAll(() => {
      global.fetch.mockRestore();
      global.M.toast.mockRestore();
      delete global.fetch;
      delete global.M.toast;
      delete global.M;
    });

    it('should handle error on discover request', (done) => {
      // When
      component.find(Button).at(0).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenLastCalledWith('/api/hue/discover', { method: 'GET' });

        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();

        expect(toastSpy).toHaveBeenCalledTimes(1);
        expect(toastSpy).toHaveBeenLastCalledWith({ html: errorMessage, classes: 'red darken-4' });

        done();
      });
    });

    it('should handle error on connect request', (done) => {
      // Given
      component.setState({ ipaddress, token: null });

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        // expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenLastCalledWith('/api/hue/connect', { method: 'post', headers: { Accept: 'application/json', 'Content-Type': 'application/json' }, body: '{"ipaddress":"192.168.1.10"}' });

        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();

        expect(toastSpy).toHaveBeenCalledTimes(1);
        expect(toastSpy).toHaveBeenLastCalledWith({ html: errorMessage, classes: 'red darken-4' });

        done();
      });
    });

    it('should handle error on info request', (done) => {
      // Given
      component.setState({ ipaddress, token: connectData.token });

      // When
      component.find(Button).at(1).simulate('click');

      // Then
      expect(setStateSpy).toHaveBeenCalledTimes(1);
      expect(component.state().isLoading).toBeTruthy();
      setImmediate(() => {
        expect(global.fetch).toHaveBeenCalledTimes(2);
        expect(global.fetch).toHaveBeenNthCalledWith(1, roomsEndpoint, { method: 'get', headers: expectedHeaders });
        expect(global.fetch).toHaveBeenNthCalledWith(2, lightsEndpoint, { method: 'get', headers: expectedHeaders });

        expect(setStateSpy).toHaveBeenCalledTimes(2);
        expect(component.state().isLoading).toBeFalsy();

        expect(toastSpy).toHaveBeenCalledTimes(1);
        expect(toastSpy).toHaveBeenLastCalledWith({ html: errorMessage, classes: 'red darken-4' });
        done();
      });
    });
  });
});
