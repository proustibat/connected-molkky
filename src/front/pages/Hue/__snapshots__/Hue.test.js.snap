// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Hue Given the api requests succeed should handle connect request: Render After Connect 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <h1>
      Hello World!
    </h1>
    <p
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        style={
          Object {
            "marginRight": "1rem",
            "textAlign": "justify",
          }
        }
      >
        Bridge IP detected: 192.168.1.10
      </span>
      <Button
        className=""
        disabled={false}
        onClick={[Function]}
        size="btn-small"
        style={
          Object {
            "display": "inline-block",
            "margin": 0,
            "width": "auto",
          }
        }
      >
        Re-scan
      </Button>
    </p>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      size="btn-large"
      style={Object {}}
    >
      Get information about your home
    </Button>
  </div>
</div>
`;

exports[`Page Hue Given the api requests succeed should handle discover request: Render After Discover 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <h1>
      Hello World!
    </h1>
    <p
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        style={
          Object {
            "marginRight": "1rem",
            "textAlign": "justify",
          }
        }
      >
        Bridge IP detected: 192.168.1.10
      </span>
      <Button
        className=""
        disabled={false}
        onClick={[Function]}
        size="btn-small"
        style={
          Object {
            "display": "inline-block",
            "margin": 0,
            "width": "auto",
          }
        }
      >
        Re-scan
      </Button>
    </p>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      size="btn-large"
      style={Object {}}
    >
      Connect
    </Button>
  </div>
</div>
`;

exports[`Page Hue Given the api requests succeed should handle info request 1`] = `
Array [
  "You have 3 rooms:",
  "You have 3 lights:",
]
`;

exports[`Page Hue Given the api requests succeed should handle info request: Render After Info Request 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <h1>
      Hello World!
    </h1>
    <p
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        style={
          Object {
            "marginRight": "1rem",
            "textAlign": "justify",
          }
        }
      >
        Bridge IP detected: 192.168.1.10
      </span>
      <Button
        className=""
        disabled={false}
        onClick={[Function]}
        size="btn-small"
        style={
          Object {
            "display": "inline-block",
            "margin": 0,
            "width": "auto",
          }
        }
      >
        Re-scan
      </Button>
    </p>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      size="btn-large"
      style={Object {}}
    >
      Get information about your home
    </Button>
    <List
      elements={
        Array [
          "chambre",
          "salon",
          "salle de bain",
        ]
      }
      title="You have 3 rooms:"
    />
    <List
      elements={
        Array [
          "bedroom-main",
          "living-room-main",
          "bathroom-main",
        ]
      }
      title="You have 3 lights:"
    />
  </div>
</div>
`;

exports[`Page Hue Given the api requests succeed should render the information lists correctly when there is no element 1`] = `
Array [
  "You have 0 room",
  "You have 0 light",
]
`;

exports[`Page Hue Given the api requests succeed should render the information lists correctly when there is only one element 1`] = `
Array [
  "You have 1 room:",
  "You have 1 light:",
]
`;

exports[`Page Hue should render the page correctly on init 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      size="btn-large"
      style={Object {}}
    >
      Scan Philips Hue System
    </Button>
  </div>
</div>
`;

exports[`Page Hue should render the page correctly with all props and state data 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <h1>
      Hello World!
    </h1>
    <p
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        style={
          Object {
            "marginRight": "1rem",
            "textAlign": "justify",
          }
        }
      >
        Bridge IP detected: 192.168.1.10
      </span>
      <Button
        className=""
        disabled={false}
        onClick={[Function]}
        size="btn-small"
        style={
          Object {
            "display": "inline-block",
            "margin": 0,
            "width": "auto",
          }
        }
      >
        Re-scan
      </Button>
    </p>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      size="btn-large"
      style={Object {}}
    >
      Get information about your home
    </Button>
    <List
      elements={
        Array [
          "chambre",
          "salon",
          "salle de bain",
        ]
      }
      title="You have 3 rooms:"
    />
    <List
      elements={
        Array [
          "bedroom-main",
          "living-room-main",
          "bathroom-main",
        ]
      }
      title="You have 3 lights:"
    />
  </div>
</div>
`;
